src/ert/_c_wrappers/enkf/ert_config.py

```diff
@@ -69,58 +75,113 @@ class ErtConfig:
     user_config_file: str = "no_config"
     config_path: str = field(init=False)
 
-    def __post_init__(self):
-        self.config_path = (
-            os.path.dirname(os.path.abspath(self.user_config_file))
-            if self.user_config_file
-            else os.getcwd()
-        )
-
     @classmethod
     def from_file(
-        cls, user_config_file, use_new_parser: bool = USE_NEW_PARSER_BY_DEFAULT
+        cls,
+        user_config_file,
+        collected_errors: List[ErrorInfo] = None,
+        use_new_parser: bool = USE_NEW_PARSER_BY_DEFAULT,
     ) -> "ErtConfig":
+        do_raise_errors = False
+        if collected_errors is None:
+            collected_errors = []
+            do_raise_errors = True
+
         user_config_dict = ErtConfig.read_user_config(
-            user_config_file, use_new_parser=use_new_parser
+            user_config_file,
+            collected_errors=collected_errors,
+            use_new_parser=use_new_parser,
         )
+
+        if not user_config_dict:
+            if do_raise_errors:
+                ConfigValidationError.raise_from_collected(collected_errors)
+            return None
+
         config_dir = os.path.abspath(os.path.dirname(user_config_file))
         ErtConfig._log_config_file(user_config_file)
+
         ErtConfig._log_config_dict(user_config_dict)
         ErtConfig.apply_config_content_defaults(user_config_dict, config_dir)
-        return ErtConfig.from_dict(user_config_dict)
+
+        the_config = ErtConfig.from_dict(user_config_dict, collected_errors)
+
+        if do_raise_errors:
+            ConfigValidationError.raise_from_collected(collected_errors)
+
+        return the_config
+
+    def __post_init__(self):
+        self.config_path = (
+            os.path.dirname(os.path.abspath(self.user_config_file))
+            if self.user_config_file
+            else os.getcwd()
+        )
 
     @classmethod
-    def from_dict(cls, config_dict) -> "ErtConfig":
+    def from_dict(
+        cls, config_dict, collected_errors: Optional[List[ErrorInfo]] = None
+    ) -> "ErtConfig":
+        do_raise_errors = False
+        if collected_errors is None:
+            collected_errors = []
+            do_raise_errors = True
```

Comment:
This logic is fairly hard to follow. It would seem to me that we should always raise errors when there are some. in this function. Also, relying on the function not raising under certain conditions is going to be ahrd to manage. It would make more sense to always raise at the end, with the option of other parts of the code raising early if there are problems from which we cannot  continue.